pkg/controller/resourcegraphdefinition/controller.go

```diff
@@ -154,7 +154,14 @@ func (r *ResourceGraphDefinitionReconciler) findRGDsForCRD(ctx context.Context,
 	}
 }
 
-func (r *ResourceGraphDefinitionReconciler) Reconcile(ctx context.Context, o *v1alpha1.ResourceGraphDefinition) (ctrl.Result, error) {
+func (r *ResourceGraphDefinitionReconciler) Reconcile(ctx context.Context, o *v1alpha1.ResourceGraphDefinition) (res ctrl.Result, reconcileErr error) {
+	var topologicalOrder []string
+	var resourcesInformation []v1alpha1.ResourceInformation
+	defer func() {
+		if err := r.updateStatus(ctx, o, topologicalOrder, resourcesInformation); err != nil {
+			reconcileErr = errors.Join(reconcileErr, err)
```

Comment:
Did  your `deferlinter` catch this @jakobmoellerdev  ?

Reply:
yeah, https://github.com/jakobmoellerdev/deferrlint spotted this:

```
deferrlint ./...
.../kro/pkg/controller/instance/controller_reconcile.go:120:6: deferred function assigns to error-typed variable `err`, but it is not a named return value
.../kro/pkg/controller/resourcegraphdefinition/controller.go:166:6: deferred function assigns to error-typed variable `err`, but it is not a named return value
```

Reply:
I will say this is a strange one and the linter doesnt detect this correctly though, I believe reconcileErr should be fine here because it is reassigned